resources/hiding_ci/build_and_install_kernel.sh

```diff
@@ -0,0 +1,173 @@
+#!/bin/bash
+# Copyright 2025 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# fail if we encounter an error, uninitialized variable or a pipe breaks
+set -eu -o pipefail
+
+check_root() {
+  # We need sudo privilleges to install the kernel
+  if [ "$(id -u)" -ne 0 ]; then
+    echo "To install, this script must be run as root or with sudo privileges"
+    exit 1
+  fi
+}
+
+check_ubuntu() {
+  # Currently this script only works on Ubuntu instances
+  if ! grep -qi 'ubuntu' /etc/os-release; then
+    echo "This script currently only works on Ubuntu."
+    exit 1
+  fi
+}
+
+tidy_up() {
+  # Some cleanup after we are done
+  popd
+  rm -rf $TMP_BUILD_DIR
+}
+
+confirm() {
+  if [[ "$*" == *"--no-install"* ]]; then
+    echo "Not installing new kernel."
+
+    if [[ "$*" == *"--tidy"* ]]; then
+      echo "Cleaning up.."
```

Comment:
nit: this echo probably belongs into `tidy_up()` xP

Reply:
Moved over now!